upgrade.go

```diff
@@ -204,12 +206,47 @@ func WithResourceProviderServer(s pulumirpc.ResourceProviderServer) Option {
 	return func(b *ProviderTest) { b.upgradeOpts.resourceProviderServer = s }
 }
 
+type Diff struct {
+	URN                 resource.URN
+	HasChanges          bool
+	Diffs               []string
+	Replaces            []string
```

Comment:
What is the format of `Diffs` and `Replaces`? Is it a `resource.PropertyPath`, or the raw gRPC diff object values (and thus arbitrary)?

Regardless, can we have that information in a comment on the field.